### PR TITLE
[12.0][ADD] purchase_order_line_price_history_discount: new module

### PR DESCRIPTION
**Pull request by ernestotejeda**
cc @Tecnativa TT24924

### CODE REVIEW

**Comment by CarlosRoca13** [comment]
When I update the product price of a product with the history, the discount is not added. I don't know if that should happen, but could be a good upgrade.

**Comment by ernestotejeda** [comment]
Thanks @CarlosRoca13 , I have talked with @pedrobaeza and we think it may be an update, so I did it. Take a look at this, please.
I also add a unit test.

File: purchase_order_line_price_history_discount/wizards/purchase_order_line_price_history.py [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+from odoo import fields, models
+
+
+class PurchaseOrderLinePriceHistoryline(models.TransientModel):
+    _inherit = "purchase.order.line.price.history.line"
+
+    discount = fields.Float(related="purchase_order_line_id.discount")
+
+    def _prepare_purchase_order_line_vals(self):
+        vals = super()._prepare_purchase_order_line_vals()
```
Comment (Tardo) [inline purchase_order_line_price_history_discount/wizards/purchase_order_line_price_history.py:12]:
```suggestion
    def _prepare_purchase_order_line_vals(self):
        self.ensure_one()
        vals = super()._prepare_purchase_order_line_vals()
```

**Review by Tardo — CHANGES_REQUESTED** [review]

File: purchase_order_line_price_history_discount/wizards/purchase_order_line_price_history.py (hunk 1 above)
Comment (ernestotejeda) [reply purchase_order_line_price_history_discount/wizards/purchase_order_line_price_history.py:12]: I think in this case it isn't necessary. When the method is called from super(), `self.ensure_one()` will be executed.
https://github.com/OCA/purchase-workflow/pull/953/files#diff-b595047d950f62cc98957ba740167de9R110

Comment (pedrobaeza) [reply purchase_order_line_price_history_discount/wizards/purchase_order_line_price_history.py:12]: I agree

**Review by CarlosRoca13 — APPROVED** [review]
LGTM :+1:

**Review by pedrobaeza — APPROVED** [review]
Code review

**Pull request merged**